include/dali/core/geom/mat.h

```diff
@@ -539,6 +540,82 @@ std::ostream &operator<<(std::ostream &os, const dali::mat<rows, cols, T> &m) {
   return os;
 }
 
+template <int n, int m, typename T>
+DALI_HOST_DEV
+inline void solve_gauss(mat<n, n, T> &A, mat<n, m, T> &B) {
```

Comment:
I would add a comment what is happening here.

Reply:
I've added a comment, not sure if it's clear enough.